feat: 신규 사용자 등록 API 구현

faber-h opened the pull request:
### ✨ 이슈 번호

- #21 

### 📌 설명

- 신규 사용자 등록 API를 구현하여 작성한 Pull Request입니다.

### 📃 작업 사항

- [x] 신규 사용자 등록 API 구현
  - [x] Request body에서 `isAdDetect`, `isReturnChannel` 값을 담아 전송
  - [x] 새로운 사용자 ID를 생성하고 사용자 정보를 DB에 저장
- [x]  성공 시
  - [x]  응답으로 `201`상태 코드와 "사용자 등록 성공했습니다.", `userId`를 반환
- [x]  실패 시
  - [x]  잘못된 요청일 경우 `400` 상태 코드와 "올바르지 않은 형식입니다."응답을 반환
  - [x]  내부 서버 오류 시 `500` 상태 코드와 "서버에서 요청을 처리할 수 없습니다."응답을 반환

### 💭 리뷰 사항 반영

- [x] `caseConverter.js를` 확장
  - [x] 카멜 케이스에서 스네이크 케이스로 변환하는 함수(toSnakeCase)
  - [x] 문자 카멜 케이스를 문자 스네이크 케이스로 변환하는 함수(stringToSnakeCase)
- [x] createUser 로직 stringToSnakeCase 활용해서 리팩토링

### ✅ Pull Request 체크 사항

- [x] 가장 최신 브랜치를 pull 하였습니다.
- [x] base 브랜치명을 확인하였습니다.
- [x] 코드 컨벤션을 모두 확인하였습니다.
- [x] 브랜치명을 확인하였습니다.

faber-h committed:
feat: Supabase에서 신규 사용자 등록 서비스 로직 구현

faber-h committed:
feat: 신규 사용자를 등록하는 컨트롤러 구현

faber-h committed:
feat: 신규 사용자 등록 API 라우터 index.js 추가

faber-h committed:
feat: 신규 사용자 등록 라우터 등록 (/users)

devjaeyoon commented on src/controllers/userController.js:
```diff
@@ -22,6 +23,27 @@ export const getUserById = async (req, res, next) => {
 
+export const createUser = async (req, res, next) => {
+  try {
+    const { isAdDetect = true, isReturnChannel = false } = req.body || {};
```
서비스의 흐름이 메인 화면에 접근하면 자동으로 (사용자 정보가 없으면)신규 사용자 등록을 진행하는 방식인데 body에 옵션을 굳이 넣어주지 않아도 될 것 같다 생각해요. 희주님은 어떻게 생각하시나요?

faber-h replied:
맞아요!
사용자 생성할 때 동작을 보장하기 위해서 `default` 값을 지정해줬는데
재윤님 말처럼 `DB`가 `default` 값을 이미 가지고 있으니 굳이 넣어주지 않아도 될 것 같네요.
좋은 의견 감사합니다~

devjaeyoon commented on src/server.js:
```diff
@@ -5,7 +5,8 @@ import { errorHandler } from "./middlewares/errorHandler.js";
 import radioRoutes from "./routes/radio-channels/allRadioList.js";
-import userRoutes from "./routes/users/[userId].js";
+import userIdRoutes from "./routes/users/[userId].js";
+import userRoutes from "./routes/users/index.js";
```
`userRoutes`와 `userIdRoutes`의
로직이 상당히 유사한 것 같은데 하나로 묶을 수는 없나요?

faber-h replied:
맞아요! 두 라우트를 하나로 통합하는 것이 가능해요.
다만 현재는 [userId].js 파일을 사용하는 브랜치 작업이 존재하기 때문에, 당장 통합하기보다는 추후 별도 이슈로 등록해 처리하는 것이 좋을 것 같아요.
좋은 의견 감사합니다!

faber-h replied:
[라우터 구조 개선 및 통합](https://github.com/Radio-Premium/RadioPremium-BE/issues/28) issue 등록 후 처리했습니다~

faber-h committed:
fix: 신규 사용자 등록 시 boolean 타입 검증 조건 수정

faber-h committed:
refactor: 신규 사용자 등록 시 body 기본값 제거
- DB 기본값 존재하기 때문에 중복으로 기본값 설정한 부분 제거
- registerUser 함수 파라미터를 객체 형태로 변경
- 오타 수정

devjaeyoon commented on src/controllers/userController.js:
```diff
@@ -22,6 +23,34 @@ export const getUserById = async (req, res, next) => {
 };
 
+export const createUser = async (req, res, next) => {
+  try {
```
지금 진행하고 계신 작업이 카멜 케이스에서 스네이크 케이스로 변환하는 작업인 것 같아요! 저희 `caseConverter.js`에 있는 `toCamelCase`는 스네이크 케이스에서 카멜 케이스로 변환만 해주고 있어서 카멜 케이스에서 스네이크 케이스를 만들어 리팩토링하는 방향으로 진행해보는 건 어떨까요?

faber-h replied:
좋은 의견 감사해요!
`caseConverter.js`를 확장해서 아래의 함수를 추가했어요.
1. 카멜 케이스에서 스네이크 케이스로 변환하는 함수(toSnakeCase)
2. 문자 카멜 케이스를 문자 스네이크 케이스로 변환하는 함수(stringToSnakeCase)

createUser 로직은 stringToSnakeCase 활용해서 리팩토링했어요~
혹시 다른 곳에서도 필요할 경우 활용하시면 될 것 같아요!

devjaeyoon replied:
감사합니다~!

faber-h committed:
refactor: 스네이크 케이스 유틸 함수 도입 및 createUser 로직 정리
- toSnakeCase 함수 작성
- stringToSnakeCase 함수 작성
- stringToSnakeCase 함수 이용하여 createUser 로직 정리

devjaeyoon reviewed (approved):
고생하셨습니다. LGTM!

Eun0713 reviewed (approved):
LGTM✨
